x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DataLifecycleServiceRuntimeSecurityIT.java

```diff
@@ -0,0 +1,274 @@
+        String dataStreamName = "metrics-foo";
+        // empty lifecycle contains the default rollover
+        prepareDataStreamAndIndex(dataStreamName, new DataLifecycle());
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(2));
+            String backingIndex = backingIndices.get(0).getName();
+            assertThat(backingIndex, backingIndexEqualTo(dataStreamName, 1));
+            String writeIndex = backingIndices.get(1).getName();
+            assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
+        });
+
+        // Index another doc to force another rollover and trigger an attempted force-merge. The force-merge may be a noop under
+        // the hood but for authz purposes this doesn't matter, it only matters that the force-merge API was called
+        indexDoc(dataStreamName);
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(3));
+        });
+    }
+
+    public void testRolloverAndRetentionAuthorized() throws Exception {
+        String dataStreamName = "metrics-foo";
+        prepareDataStreamAndIndex(dataStreamName, new DataLifecycle(TimeValue.timeValueMillis(0)));
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(1));
+            // we expect the data stream to have only one backing index, the write one, with generation 2
+            // as generation 1 would've been deleted by DLM given the lifecycle configuration
+            String writeIndex = backingIndices.get(0).getName();
+            assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
+        });
+    }
+
+    public void testUnauthorized() throws Exception {
+        // the DLM user is not authorized for this pattern since it matches the security index
+        String dataStreamName = ".security-foo";
+        prepareDataStreamAndIndex(dataStreamName, new DataLifecycle(TimeValue.timeValueMillis(0)));
+
+        assertBusy(() -> {
+            Map<String, String> indicesAndErrors = collectErrorsFromStoreAsMap();
+            assertThat(indicesAndErrors, is(not(anEmptyMap())));
+            assertThat(
+                indicesAndErrors.values(),
+                hasItem(allOf(containsString("security_exception"), containsString("unauthorized for user [_dlm]")))
+            );
+        });
+    }
+
+    public void testRolloverAndRetentionWithSystemDataStreamAuthorized() throws Exception {
+        String dataStreamName = SystemDataStreamTestPlugin.SYSTEM_DATA_STREAM_NAME;
+        indexDoc(dataStreamName);
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
```

Comment:
as mentioned earlier i don't think we should allow dlm (by default) to manage system indices

Reply:
See my comment [here](https://github.com/elastic/elasticsearch/pull/96253#discussion_r1205386573).